tests/test_models.py

```diff
@@ -133,6 +181,45 @@ class B(BaseModel, GraphQLFragmentMixin):
     """)
     assert b_fragment.name == "fragmentB"
     assert b_fragment.body == b_expect
+    assert not b_fragment.lazy
+    assert b_fragments[1] == a_fragment
+
+
+def test_graphql_fragment_mixin_lazy() -> None:
+    a_fragments = A.gql_fragments(lazy=True)
+    assert len(a_fragments) == 1
+    a_fragment = a_fragments[0]
+
+    a_expect = normalize_query("""
+    fragment fragmentA on A {
+        fooBar
+    }
+    """)
+    assert a_fragment.name == "fragmentA"
+    assert a_fragment.body == a_expect
+    assert a_fragment.lazy
+
+    b_fragments = B.gql_fragments(lazy=True)
+    assert len(b_fragments) == 2
+    b_fragment = b_fragments[0]
+
+    b_expect = normalize_query("""
+    fragment fragmentB on B {
+        helloWorld
+        baz
+        optionalStr
+        strOrInt
+        eagerA {
+            ...fragmentA
+        }
+        notLazyOptionalA {
+            ...fragmentA
+        }
```

Comment:
note how the `.gql_fragments()` method does not return `lazyA` nor `manyA` if `lazy=True`